### PR TITLE
fix(gossip,accountsdb): removed leaks in error path, constified things, pass slices instead of ArrayList

### PR DESCRIPTION
**Pull request by InKryption**
* Switch from `std.ArrayList` to `std.ArrayListUnmanaged` to save memory in some cases.

* Improve `spawnThreadTasks` to output into a caller-provided arraylist, allowing them the ability to properly clean up even in the error path. Also document more precisely.

* Use tuple destructuring feature now available in zig 0.12.

* Constify certain things not caught by zig's ast checks.

* Fix leaks.

Supercedes #142.

### CODE REVIEW

File: src/cmd/cmd.zig
Comment (0xNineteen) [inline src/cmd/cmd.zig:573]: few things:
- can we rename this to `runGossipWithConfigValues` or something more descriptive? 
- with this change we can remove all `spawnGossip` calls to just spawn the thread directly (vs single-line fn call)
- lastly, can we change `current` variable name to `gossip_config`

File: src/gossip/fuzz.zig
Comment (0xNineteen) [inline src/gossip/fuzz.zig:306]: never seen that syntax before - thats cool

File: src/utils/thread.zig
Comment (0xNineteen) [inline src/utils/thread.zig:34]: yeah thats an interesting idea - rn we only care about full coverage but could be useful in the future

**Review by 0xNineteen — CHANGES_REQUESTED** [review]
small naming changes - otherwise lgtm

File: src/cmd/cmd.zig
Comment (InKryption) [reply src/cmd/cmd.zig]: can do

File: src/gossip/fuzz.zig
Comment (InKryption) [reply src/gossip/fuzz.zig:306]: Yeah, it's semi-recent I believe. I believe it was added in 0.12.

File: src/utils/thread.zig
Comment (InKryption) [reply src/utils/thread.zig:34]: Yeah, I didn't see anywhere currently where it'd be useful, but it's not impossible - being able to continue execution with accurate information in spite of a system hiccup or lack of available memory seems valuable enough that I'd want to keep the thought salient through this TODO.

**Review by 0xNineteen — APPROVED** [review]
lgtm - i think we should wait for drew to approve also before merging

**Review by dnut — APPROVED** [review]

**Pull request merged**
